### PR TITLE
Check ENI

### PR DESCRIPTION
**Pull request by damiendurant**
#### New Plugins

- [x] Tests : rubocop ok.

- [x] Add the plugin to the README

- [x] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### Purpose

Check the status of ENI (in-use or available)

### CODE REVIEW

**Comment by majormoses** [comment]
@damiendurant thanks for the contribution, could I get you to add a unit test?

Also ideally all aws checks should support checking multiple "instances" within the same check as loading the aws-sdk is quite resource intensive. An example of this would be the sqs checks which support both a prefix or a comma separated list of names. This is based on the comments here: #24, #27

File: bin/check-eni-status.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    { access_key_id: config[:aws_access_key],
+      secret_access_key: config[:aws_secret_access_key],
+      region: config[:aws_region] }
+  end
+
+  def run
+    Aws.config.update(aws_config)
+    client = Aws::EC2::Client.new
+
+    status = client.describe_network_interfaces(filters: [{ name: 'network-interface-id', values: [config[:eni].to_s] }])[:network_interfaces].first
```
Comment (majormoses) [inline bin/check-eni-status.rb]: can you break this line up its hard to read.

File: bin/check-eni-status.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
+#! /usr/bin/env ruby
+#
+# check-eni-status
+#
+# DESCRIPTION:
+#   This plugin checks the status of an elastic number interface
```
Comment (eheydrick) [inline bin/check-eni-status.rb]: number should be network

File: bin/check-eni-status.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'sensu-plugins-aws'
+require 'aws-sdk'
+
+#
+# Check SQS Messages
```
Comment (eheydrick) [inline bin/check-eni-status.rb]: update comment for this check

File: bin/check-eni-status.rb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  option :aws_region,
+         short: '-r AWS_REGION',
+         long: '--aws-region REGION',
+         description: 'AWS Region (defaults to us-east-1).',
+         default: 'us-east-1'
+
+  option :eni,
+         short: '-e ENI_ID',
+         long: '--eni ENI_ID',
+         description: 'The Eleastic Network Interface to check',
```
Comment (eheydrick) [inline bin/check-eni-status.rb]: Elastic

**Comment by damiendurant** [comment]
Hi,
Based on your return I have updated the comment/desc typo, and update the code to allow multiple values as arguement.

**Comment by majormoses** [comment]
@damiendurant awesome I will take a look today and if its good for sure get a release out this evening.

**Comment by majormoses** [comment]
@damiendurant only thing I see now thats missing is a changelog entry. We normally require it done by the user but I can make an exception since you listened to our general feedback and we failed to point this out.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-aws/versions/6.1.0

**Pull request merged**